### PR TITLE
soc: arm: stm32wb serie low power modes substates

### PR DESCRIPTION
**Pull request by FRASTM**
Change the substate-id range from 1-3 as described in the include/power/power_state.h
this is changing the range in pm_power_state_set() of the soc/arm/st_stm32/stm32wb/power.c
The debug config will let the clocks active in STOP mode at init.
Following the https://github.com/zephyrproject-rtos/zephyr/pull/33131, the power.c part of the stm32wb is modified in the same way.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series [hunk 1]
```diff
@@ -12,11 +12,9 @@ config SOC_SERIES
 
 if PM
 config PM_DEVICE
-	bool
 	default y
 
 config STM32_LPTIM_TIMER
-	bool
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series:19]: Nitpick: A "+ minor fixes" in the commit message for this would have been nice

**Review by erwango — APPROVED** [review]

File: soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series (hunk 1 above)
Comment (FRASTM) [reply soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series:19]: added in the commit msg

**Comment by FRASTM** [comment]
rebase and change the substate range

**Review by ABOSTM — CHANGES_REQUESTED** [review]
Commit Message:
> . . .
> The low power states 0∕1∕2 are added to the stm32wb55rg
> nucleo board, with (default min residency values).

Sentence not finished: With what ? (or wrong parenthesis)

Otherwise LGTM

**Comment by FRASTM** [comment]
rebase
changed info.substate_id in the pm_power_state_exit_post_ops function

**Review by ABOSTM — APPROVED** [review]
LGTM

**Pull request merged**